spec/formatting.md

```diff
@@ -247,6 +238,37 @@ the following steps are taken:
    If the call fails or does not return a valid value,
    emit a Resolution error and use a _fallback value_ for the _expression_.
 
+#### Option Resolution
+
+The result of resolving _option_ values is a mapping of string identifiers to values.
+
+For each _option_:
+
+- Resolve the _identifier_ of the _option_.
+- If the _option_'s _identifier_ already exists in the resolved mapping of _options_,
+   emit a Duplicate Option Name error.
+- If the _option_'s right-hand side successfully resolves to a value,
+   bind the _identifier_ of the _option_ to the resolved value in the mapping.
+- Otherwise, bind the _identifier_ of the _option_ to an unresolved value in the mapping.
+   Implementations MAY later remove this value before calling the _function_.
+   (Note that an Unresolved Variable error will have been emitted.)
+
+Errors MAY be emitted during _option resolution_,
+but it always resolves to some mapping of string identifiers to values.
+This mapping can be empty.
+
+### Markup Resolution
+
+Unlike _functions_, the resolution of _markup_ is not customizable.
+
+The resolved value of _markup_ includes the following fields:
+
+- The type of the markup: open, standalone, or close
+- The _identifier_ of the _markup_
+- For _markup-open_ and _markup_standalone_,
+  the resolved _options_ values after _option resolution_.
+
+The resolution of _markup_ MUST always succeed.
```

Comment:
This doesn't look quite right for a use of RFC 2119 "MUST".

Reply:
The usage here is intentional, at least on my part: An implementation must not allow for the resolution of any `{#foo}` part to ever fail.